src/Identity/Extensions.Core/src/IdentityBuilder.cs

```diff
@@ -21,6 +22,7 @@ public class IdentityBuilder
     /// <param name="services">The <see cref="IServiceCollection"/> to attach to.</param>
     public IdentityBuilder(Type user, IServiceCollection services)
     {
+        Debug.Assert(user.IsClass);
```

Comment:
I can new up my own IdentityBuilder with an arbitrary type with no validation, right? I know that might be weird, but this is public. It looks like `AddIdentityCore<TUser>` doesn't validate either.

If there's any way whatsoever to make a `Debug.Assert` fail without resorting to something like private reflection, it should not be an assert. It should be a normal runtime check that throws for value types.

Reply:
You're right. Fortunately, there are class constraints everywhere on the types that user and role types are used with, so this would always fail, just not at this point. I'll add runtime validation.